lib/pages/home/pages/profile/widgets/profile_card/widgets/profile_details/widgets/profile_age.dart

```diff
@@ -0,0 +1,96 @@
+import 'package:Okuna/models/user.dart';
+import 'package:Okuna/services/localization.dart';
+import 'package:Okuna/services/toast.dart';
+import 'package:Okuna/services/utils_service.dart';
+import 'package:Okuna/widgets/icon.dart';
+import 'package:Okuna/widgets/theming/text.dart';
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+import '../../../../../../../../../provider.dart';
+
+class OBProfileAge extends StatefulWidget {
+  final User user;
+
+  OBProfileAge(this.user);
+
+  @override
+  State<StatefulWidget> createState() {
+    return OBProfileAgeState();
+  }
+}
+
+class OBProfileAgeState extends State<OBProfileAge> {
+  LocalizationService _localizationService;
+  ToastService _toastService;
+  UtilsService _utilsService;
+  bool _needsBootstrap;
+  static const MIN_AGE_IN_YEARS_FOR_ADULT = 1;
+
+  @override
+  void initState() {
+    super.initState();
+    _needsBootstrap = true;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    var openbookProvider = OpenbookProvider.of(context);
+    _utilsService = openbookProvider.utilsService;
+    _localizationService = openbookProvider.localizationService;
+    _toastService = openbookProvider.toastService;
+    DateTime age = widget.user.dateJoined;
+
+    if (age == null) {
+      return const SizedBox();
+    }
+
+    if (_needsBootstrap) {
+      _bootstrap();
+      _needsBootstrap = false;
```

Comment:
Retrieve the openbookProvider an init services inside this block. Otherwise on every rebuild, these get re-retrieved and the .of function searches in a widget tree, so better to use it sparingly.